lib/core/checkCoreSyncFinishedFactory.js

```diff
@@ -0,0 +1,48 @@
+const wait = require('../util/wait');
+
+/**
+ * Check that core is synced (factory)
+ *
+ * @param {RpcClient} coreRpcClient
+ *
+ * @returns {checkCoreSyncFinished}
+ */
+function checkCoreSyncFinishedFactory(coreRpcClient) {
+  /**
+   * Check that core is synced
+   *
+   * @typedef checkCoreSyncFinished
+   *
+   * @param {function(number, number)} progressCallback
+   *
+   * @returns {Promise<Boolean>}
+   */
+  async function checkCoreSyncFinished(progressCallback) {
+    while (true) {
+      const {
```

Comment:
I would move this call under `isBlockchainSynced` check 'coz if it's sync then there is no progress :)